src/doc.d

```diff
@@ -2479,6 +2499,73 @@ extern (C++) void highlightCode(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t o
                 i = j - 1;
             }
         }
+        else if (!resolvedTemplateParameters)
+        {
+            // hunt for template declarations:
+            for (size_t symi = 0; symi < a.dim; symi++)
+            {
+                FuncDeclaration fd = (*a)[symi].isFuncDeclaration();
+
+                if (!fd || !fd.parent || !fd.parent.isTemplateDeclaration())
+                {
+                    continue;
+                }
+
+                TemplateDeclaration td = fd.parent.isTemplateDeclaration();
+
+                // build the template parameters
+                size_t[] paramLens;
+                OutBuffer parametersBuf;
+                HdrGenState hgs;
+                parametersBuf.writeByte('(');
+
+                for (size_t parami = 0; parami < td.parameters.dim; parami++)
+                {
+                    TemplateParameter tp = (*td.parameters)[parami];
+
+                    if (parami)
+                        parametersBuf.writestring(", ");
+
+                    size_t lastOffset = parametersBuf.offset;
+
+                    .toCBuffer(tp, &parametersBuf, &hgs);
+
+                    paramLens ~= [parametersBuf.offset - lastOffset];
+                }
+                parametersBuf.writeByte(')');
+
+                const(char*) templateParams = parametersBuf.extractString();
+                size_t templateParamsLen = strlen(templateParams);
+
+                //printf("templateDecl: %s\ntemplateParams: %s\nstart: %s\n", td.toChars(), templateParams, start);
+
+                if (cmp(templateParams, start, templateParamsLen) == 0)
+                {
+                    const(char*) templateParamListMacro = "$(DDOC_TEMPLATE_PARAM_LIST ";
+                    size_t paramListEnd = buf.bracket(i, templateParamListMacro, i + templateParamsLen, ")") - 1;
+
+                    // We have the parameter list. While we're here we might
+                    // as well wrap the parameters themselves as well
+
+                    // + 1 here to take into account the opening paren of the
+                    // template param list
+                    i += strlen(templateParamListMacro) + 1;
```

Comment:
Better make `templateParamListMacro` an `immutable` and then use `templateParamListMacro.length` here. When you need the pointer you use `.ptr`. Also that will also be zero terminated.